src/cascadia/TerminalControl/KeyChord.cpp

```diff
@@ -38,19 +38,68 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         {
             _vkey = MapVirtualKeyW(scanCode, MAPVK_VSC_TO_VK_EX);
         }
+
+        assert(_vkey || _scanCode);
+    }
+
+    uint64_t KeyChord::Hash() const noexcept
+    {
+        // Two KeyChords are equal if they have the same modifiers and either identical
+        // Vkey or ScanCode, with Vkey being preferred. See KeyChord::Equals().
+        // This forces us to _either_ hash _vkey or _scanCode.
+        //
+        // Additionally the hash value with _vkey==123 and _scanCode==123 must be different.
+        // --> Mark the hash of a KeyChord without _vkey with the highest bit set,
+        //     thus generating a different Hash() even if the Vkey/ScanCode are identical.
+        auto h = static_cast<uint64_t>(_modifiers) << 32;
```

Comment:
qq: since x86 has `4==sizeof(size_t)`, will this cause us to ignore modifiers when we cast off the top 32 bits and stuff these in the map? Do we need to collapse this down to 32 bits total?

Reply:
A hashmap would even work if the hash function returns the same value for all keys.
While this hash function is very primitive, it's still good enough that cutting off the top 32 bits will not loose too much meaningful information. The modifiers in particular won't be cut off since those are already mixed in and distributed using our hash function.

Reply:
oh right. thanks!